core/domain/exp_jobs_one_off_test.py

```diff
@@ -2360,3 +2366,241 @@ def test_no_action_is_performed_for_deleted_exploration(self):
+                job_id)
+            )
+        expected_output = [('[u\'State schema version is '
+                            'not the minimum version expected\', '
+                            '[u\'exp_id0\']]')]
+        self.assertEqual(actual_output, expected_output)
+
+    def test_duration_secs_is_updated_when_job_is_complete(self):
+        exploration = exp_domain.Exploration.create_default_exploration(
+            self.VALID_EXP_ID, title='Exploration Title 1', category='category')
+        exploration.add_states(['State1'])
+        state1 = exploration.states['State1']
+        recorded_voiceovers_dict = {
+            'voiceovers_mapping': {
+                'content': {
+                    'en': {
+                        'filename': self.TEST_AUDIO_FILE_MP3,
+                        'file_size_bytes': 123,
+                        'needs_update': True,
+                        'duration_secs': 0.0
+                    },
+                    'hi': {
+                        'filename': self.TEST_AUDIO_FILE_MP3,
+                        'file_size_bytes': 1234,
+                        'needs_update': False,
+                        'duration_secs': 0.0
+                    }
+                },
+                'default_outcome': {}
+            }
+        }
+
+        # Get the test audio file and prepare for storage.
+        with python_utils.open_file(
+            os.path.join(feconf.TESTS_DATA_DIR, self.TEST_AUDIO_FILE_MP3),
+            mode='rb', encoding=None) as f:
+            raw_audio = f.read()
+
+        tempbuffer = python_utils.string_io()
+        tempbuffer.write(raw_audio)
+        tempbuffer.seek(0)
+
+        audio = mp3.MP3(tempbuffer)
+
+        tempbuffer.close()
+        mimetype = audio.mime[0]
+        del audio
+
+        # Save the raw audio into the filesystem.
+        file_system_class = fs_services.get_entity_file_system_class()
+        fs = fs_domain.AbstractFileSystem(file_system_class(
+            feconf.ENTITY_TYPE_EXPLORATION, self.VALID_EXP_ID))
+        fs.commit(
+            '%s/%s' % (self._FILENAME_PREFIX, self.TEST_AUDIO_FILE_MP3),
+            raw_audio, mimetype=mimetype)
+
+        # Save the recorded_voiceovers.
+        state1.update_recorded_voiceovers(
+            state_domain.RecordedVoiceovers.from_dict(recorded_voiceovers_dict))
+        exp_services.save_new_exploration(self.user_a_id, exploration)
+
+        # Run the job.
+        job_id = (
+            exp_jobs_one_off.VoiceoverDurationSecondsOneOffJob.create_new())
+        exp_jobs_one_off.VoiceoverDurationSecondsOneOffJob.enqueue(job_id)
+        self.process_and_flush_pending_tasks()
+
+        actual_output = (
+            exp_jobs_one_off.VoiceoverDurationSecondsOneOffJob.get_output(
+                job_id)
+            )
+        # Validate job ran successfully.
+        expected_output = ['[u\'SUCCESS\', 2]']
+        self.assertEqual(actual_output, expected_output)
+
+
+        updated_model = (
```

Comment:
this is going to be a dict not model, right?

Reply:
Later on, I access this model to extract the dict for the assertion. I will either change my naming of the variable or find another way to access the saved dict data.

Reply:
Ditto

Reply:
Done.